packages/vue-ds/src/components/Pagination/Pagination.vue

```diff
@@ -0,0 +1,145 @@
+<template>
+  <section v-if="!isInvalid" :class="PaginationClass">
+    <div :class="PaginationContainerClass">
+      <button
+        v-if="hasPreviousCondition"
+        class="paginationButton"
+        :disabled="isLoading"
+        @click="prevPage"
+      >
+        <AngleLeft :width="12" :height="23" :viewBox="'-1 3 12 7'" />
+      </button>
+      <div
+        v-for="(currentPage, idx) in range"
+        :key="idx"
+        class="indexWrapper"
+        :class="page === currentPage ? 'active' : ''"
+        :data-marker="page === currentPage && `de ${count}`"
+      >
+        <div class="paginationIndex" @click="selectPage(currentPage)">
+          <p>{{ currentPage }}</p>
+        </div>
+      </div>
+      <button
+        v-if="hasNextCondition"
+        class="paginationButton"
+        :disabled="isLoading"
+        @click="nextPage"
+      >
+        <AngleRight :width="12" :height="23" :viewBox="'-3 3 12 7'" />
+      </button>
+    </div>
+  </section>
+</template>
+
+<script lang="ts">
+import { Component, Vue, Prop, Watch } from 'vue-property-decorator'
+
+import {
+  getRangeIndexes,
+  hasNextRange,
+  hasPreviousRange,
+} from '@venice/core/helpers/Pagination'
+import { IPagination } from '@venice/core/models'
+
+// TODO: these icons should be replaced when there is an icon library
+import AngleLeft from './icons/AngleLeft.vue'
+import AngleRight from './icons/AngleRight.vue'
+
+@Component({
+  components: {
+    AngleLeft,
+    AngleRight,
+  },
+})
+export default class Pagination extends Vue {
+  @Prop({ default: 1 }) count!: IPagination['count']
+  @Prop({ default: 1 }) page!: IPagination['page']
+  @Prop({ default: false }) isLoading!: IPagination['isLoading']
+  @Prop() onChange!: IPagination['onChange']
+  @Prop() customClass!: IPagination['className']
+
+  private maxPhoneWidth = 420
+  private isMobile = false
+  private range: number[] = []
+
+  debounce = (func: () => void, wait: number) => {
+    let timer: number
+
+    return () => {
+      clearTimeout(timer)
+      timer = window.setTimeout(func, wait)
+    }
+  }
+
+  checkDevice = () => window.innerWidth <= this.maxPhoneWidth
+
+  @Watch('page')
```

Comment:
I think this `watch` is unnecessary... 🤔🤔 it should work without him. Please, let me know if I got it wrong

Reply:
I need to call the `getRange` function when `page` props is changed to get new range for pagination. When I tried to remove the Watch decorator it doesn't work because the `created` function isn't called again when props change. Idk if this approach is correct :/ what do you think?

Reply:
Awesome! now I got it. hahaha  You are right!